File: app/src/main/java/com/miandroidchallenge/ucoppp/miandroidchallenge/di/AppModule.java
package com.miandroidchallenge.ucoppp.miandroidchallenge.di;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    private Application application;

    AppModule(Application application) {
        this.application = application;
    }

    @Provides
    @Singleton
    Application provideApplication() {
        return this.application;
    }
}

File: app/src/main/java/com/miandroidchallenge/ucoppp/miandroidchallenge/application/AppModule.kt
package com.miandroidchallenge.ucoppp.miandroidchallenge.application

import android.app.Application
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
internal class AppModule(private val mApplication: Application) {

    @Provides
    @Singleton
    fun provideApplication(): Application {
        return mApplication
    }
}